src/class-wpml-gutenberg-integration-factory.php

```diff
@@ -31,27 +31,20 @@ public function create() {
 			)
 		);
 
-		$reusable_blocks             = new WPML\PB\Gutenberg\ReusableBlocks\Blocks();
-		$reusable_blocks_translation = new WPML\PB\Gutenberg\ReusableBlocks\Translation( $sitepress );
-
-		$integrations->add(
-			new WPML\PB\Gutenberg\ReusableBlocks\Integration( $reusable_blocks_translation )
+		$is_reusable_block_translatable = $sitepress->is_translated_post_type(
+			WPML\PB\Gutenberg\ReusableBlocks\Translation::POST_TYPE
 		);
 
-		if ( is_admin() ) {
+		if ( $is_reusable_block_translatable ) {
 			$integrations->add(
```

Comment:
@strategio Do you need to share `WPML\PB\Gutenberg\ReusableBlocks\Blocks` and `WPML\PB\Gutenberg\ReusableBlocks\Translation` like you do in the original code?

eg.
```php
WPML\Container\share( [
    'WPML\PB\Gutenberg\ReusableBlocks\Blocks',
   'WPML\PB\Gutenberg\ReusableBlocks\Translation',
]);
```

Reply:
@brucepearson, sharing is not really needed because the objects don't have any specific state. But we can save some memory (and I keep forgetting to use this `share` function). That's fixed!

Reply:
@strategio I only mentioned it so you are aware of it. I only recommend sharing when it's required and not just to save a few bytes. Now it could be misleading because someone reading the code might assume that it's a requirement that there's only ever one instance of these classes.

Reply:
@brucepearson, in some DIC the default is that instances are shared.

Do you think that we should **not share** when it's not explicitly required? I can remove it if you prefer.

Reply:
Removed the class instance sharing!